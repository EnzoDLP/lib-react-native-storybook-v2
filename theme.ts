export const Colors = {
  light: {
    text: {
      primary: '#0F0F0F',
      secondary: '#000000',
      disabled: '#686767',
      onBackground: '#FFFFFF',
      link: '#008859',
      linkHover: '#22543A',
      error: '#D21E1E',
    },
    buttons: {
      primary: '#008859',
      hover: '#22543A',
      disabled: '#E7E7E7',
      onBackground: '#FFFFFF',
    },
    border: {
      primary: '#008859',
    },
    background: {
      light: '#F5F5F5',
      dark: '#686767',
      green: '#E6F3EE',
      default: '#FFFFFF',
      grey: '#e8e8e8',
    },
    input: {
      primary: {
        background: '#FFFFFF',
        border: '#686767',
      },
      focus: {
        background: '#FFFFFF',
        border: '#078213',
      },
      error: {
        background: '#FFE9ED',
        border: '#D21E1E',
      },
      disabled: {
        background: '#E7E7E7',
        border: '#B6B6B6',
      },
    },
  },
};
