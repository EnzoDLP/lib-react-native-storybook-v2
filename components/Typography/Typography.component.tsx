import { Text, TextProps } from 'react-native';
import { Colors } from '../../theme';

type TypographyVariant = 'large' | 'medium' | 'small' | 'extraSmall';
type FontWeight = 'bold' | 'semiBold' | 'regular';
type TextColor = keyof typeof Colors.light.text;

interface TypographyProps extends TextProps {
  variant?: TypographyVariant;
  weight?: FontWeight;
  color?: TextColor;
  children: React.ReactNode;
}

const typographyStyles = {
  large: { fontSize: 18, lineHeight: 24 },
  medium: { fontSize: 16, lineHeight: 22 },
  small: { fontSize: 14, lineHeight: 20 },
  extraSmall: { fontSize: 12, lineHeight: 18 },
};

const fontWeights: Record<FontWeight, { fontWeight: '400' | '600' | '700' }> = {
  bold: { fontWeight: '700' },
  semiBold: { fontWeight: '600' },
  regular: { fontWeight: '400' },
};

const Typography: React.FC<TypographyProps> = ({
  variant = 'medium',
  weight = 'regular',
  color = 'primary',
  style,
  children,
  ...props
}) => {
  return (
    <Text
      style={[
        typographyStyles[variant],
        fontWeights[weight],
        style,
        { color: Colors.light.text[color] },
      ]}
      {...props}
    >
      {children}
    </Text>
  );
};

export default Typography;
