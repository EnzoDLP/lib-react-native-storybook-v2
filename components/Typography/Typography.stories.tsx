import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { View } from 'react-native';
import Typography from './Typography.component';
import { Colors } from '../../theme';

const meta = {
  title: 'Components/Typography',
  component: Typography,
  decorators: [
    (Story) => (
      <View style={{ padding: 16 }}>
        <Story />
      </View>
    ),
  ],
  argTypes: {
    variant: {
      control: 'select',
      options: ['large', 'medium', 'small', 'extraSmall'],
    },
    weight: {
      control: 'select',
      options: ['bold', 'semiBold', 'regular'],
    },
    color: {
      control: 'select',
      options: Object.keys(Colors.light.text),
    },
    children: {
      control: 'text',
    },
  },
} satisfies Meta<typeof Typography>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    variant: 'small',
    weight: 'bold',
    color: 'error',
    children: 'Test',
  },
};
