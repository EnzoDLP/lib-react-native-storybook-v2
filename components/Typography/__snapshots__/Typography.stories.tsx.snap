// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Typography Default smoke-test 1`] = `
<div class="css-view-175oi2r"
     style="padding: 16px;"
>
  <div dir="auto"
       class="css-text-146c3p1"
       style="font-size: 14px; line-height: 20px; font-weight: 700; color: rgb(210, 30, 30);"
  >
    Test
  </div>
</div>
`;
